ARCH.md

```diff
@@ -0,0 +1,49 @@
+Rust-Lightning is broken into a number of high-level structures with various APIs to hook
+them together, as well as a number of APIs for you to provide external data which you,
+the user, have to provide.
+
+The two most important structures which nearly every application of Rust-Lightning will
+need to use are ChannelManager and ChannelMonitor. ChannelManager holds multiple Channels,
+routes payments between them and exposes a simple API to make and receive payments.
+Individual ChannelMonitors monitor the on-chain state of a Channel and punish
+counterparties if they misbehave, with the ManyChannelMonitor API providing a way for you
+to receive ChannelMonitorUpdates and persist them to disk before the Channel steps forward.
```

Comment:
This sentences is conveying a bit much. Break it into two sentences from "with the ManyChannelMonitor API ...". Since this is the first time ManyChannelMonitor is introduced, use the second also to describe how it relates to ChannelManager and ChannelMonitor, as it isn't clear.